src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/table.svelte

```diff
@@ -81,10 +96,64 @@
             show: selected?.includes(attribute.key) ?? true
         }))
     );
+
+    let selectedDb: string[] = [];
+    let showDelete = false;
+    let deleting = false;
+
+    async function handleDelete() {
+        showDelete = false;
+
+        const promises = selectedDb.map((documentId) =>
+            sdk.forProject.databases.deleteDocument(databaseId, collectionId, documentId)
+        );
+        try {
+            await Promise.all(promises);
+            trackEvent(Submit.DocumentDelete);
+            addNotification({
+                type: 'success',
+                message: `${selected.length} document${selected.length > 1 ? 's' : ''} deleted`
+            });
+            invalidate(Dependencies.DOCUMENTS);
+        } catch (error) {
+            addNotification({
+                type: 'error',
+                message: error.message
+            });
+            trackError(error, Submit.DocumentDelete);
+        } finally {
+            selected = [];
+            showDelete = false;
+        }
+    }
+
+    enum Deletion {
+        'setNull' = 'Set document ID as NULL in all related documents',
+        'cascade' = 'All related documents will be deleted',
+        'restrict' = 'Document can not be deleted'
+    }
+
+    $: relAttributes = $attributes?.filter((attribute) =>
+        isRelationship(attribute)
+    ) as Models.AttributeRelationship[];
+
+    let checked = false;
 </script>
 
 <TableScroll isSticky>
     <TableHeader>
+        <TableCellHead width={10}>
+            <InputCheckbox
+                id="select-all"
+                indeterminate={selectedDb.length > 0 && selectedDb.length < data.documents.total}
+                value={selectedDb.length === data.documents.total}
+                on:click={(e) => {
+                    if (!isHTMLInputElement(e.target)) return;
+                    selectedDb = e.target.checked
+                        ? data.documents.documents.map((database) => database.$id)
+                        : [];
+                }} />
+        </TableCellHead>
```

Comment:
@TGlide we repeat this code (or something similar) a lot... is it possible to create a custom `TableCellHeadCheck` this way it's also a lot more maintainable

Reply:
I think it makes sense to abstract the onClick fn, but not the rest of the cmp. Otherwise we'll be basically copying the InputCheckbox almost 1:1. Wdyt?